File: web/admin/src/api/location.ts
import client from './client'

export interface Location {
  id: number
  club_id: number
  name: string
  address: string
  map_url: string
  created_at?: string
  updated_at?: string
}

export interface LocationCreate {
  name: string
  address: string
  map_url: string
}

export interface LocationUpdate {
  name?: string
  address?: string
  map_url?: string
}

export default {
  list(clubId: number) {
    return client.get<Location[]>(`/clubs/${clubId}/locations`)
  },

  create(clubId: number, data: LocationCreate) {
    return client.post<Location>(`/clubs/${clubId}/locations`, data)
  },

  get(id: number) {
    return client.get<Location>(`/locations/${id}`)
  },

  update(id: number, data: LocationUpdate) {
    return client.put<Location>(`/locations/${id}`, data)
  },

  remove(id: number) {
    return client.delete(`/locations/${id}`)
  },
}

File: internal/domain/entity/location.go
package entity

import "time"

type Location struct {
	ID          int64     `json:"id"          db:"id"`
	ClubID      int64     `json:"club_id"     db:"club_id"`
	Name        string    `json:"name"        db:"name"`
	Address     string    `json:"address"     db:"address"`
	MapURL      string    `json:"map_url"     db:"map_url"`
	Description string    `json:"description" db:"description"`
	CreatedAt   time.Time `json:"created_at"  db:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"  db:"updated_at"`
}
